src/ert/scheduler/driver.py

```diff
@@ -61,6 +63,12 @@ async def poll(self) -> None:
     async def finish(self) -> None:
         """make sure that all the jobs / realizations are complete."""
 
+    def read_output_files(
```

Comment:
If this is to be used in "case of failure", it should be reflected in the function name. 

The driver could have a specific API for stdout and stderr, perhaps a failure message code can be made driver independent based on stdout/stderr api?